lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -412,6 +412,42 @@ getLocationInfoAsString(const SmallPtrSet<Operation *, 8> &ops) {
   return sstr.str();
 }
 
+/// Given string \p origName, generate a new name if it conflicts with any
+/// keyword or any other name in the set \p recordNames. Use the int \p
+/// nextGeneratedNameID as a counter for suffix. Update the \p recordNames with
+/// the generated name and return the StringRef.
+StringRef resolveKeywordConflict(StringRef origName,
+                                 llvm::StringSet<> &recordNames,
+                                 size_t &nextGeneratedNameID) {
+  auto name = origName;
+  // Get the list of reserved words we need to avoid.  We could prepopulate this
+  // into the used words cache, but it is large and immutable, so we just query
+  // it when needed.
+  auto &reservedWords = getReservedWords();
+  SmallVector<char, 16> nameBuffer(name.begin(), name.end());
+  nameBuffer.push_back('_');
+  auto baseSize = nameBuffer.size();
+
+  while (1) {
+    // Loop until we get a name that is not a keyword and is unique.
+    if (!reservedWords.count(name)) {
+      auto itAndInserted = recordNames.insert(name);
+      if (itAndInserted.second)
+        return itAndInserted.first->getKey();
+    }
+    // We need to auto-unique it.
+    auto suffix = llvm::utostr(nextGeneratedNameID++);
+    nameBuffer.append(suffix.begin(), suffix.end());
+    name = StringRef(nameBuffer.data(), nameBuffer.size());
+
+    // Chop off the suffix and try again until we get a unique name..
+    nameBuffer.resize(baseSize);
+  }
+
+  // Control never reaches this return.
+  return origName;
```

Comment:
It's safe to remove this return.  Some compilers will warn about this as unreachable code.

Reply:
Done